src/nspawn/nspawn.c

```diff
@@ -1074,6 +1074,15 @@ static int parse_argv(int argc, char *argv[]) {
         if (getenv_bool("SYSTEMD_NSPAWN_MOUNT_RW") <= 0)
                 mount_settings |= MOUNT_RO;
 
+        if (arg_userns_mode != USER_NAMESPACE_NO)
+                mount_settings |= MOUNT_USE_USERNS;
+
+        if (arg_private_network)
+                mount_settings |= MOUNT_USE_NETNS;
+
+        if (getenv_bool("SYSTEMD_NSPAWN_USE_NETNS") <= 0)
+                mount_settings |= MOUNT_USE_NETNS;
```

Comment:
`--private-network` creates its own network namespace, which we would like to omit here. The use case is rkt+an external network configurator (e.g. CNI), where we already have an external network namespace, join it, and have `/proc/sys/net` read-write.